tests/Service/Parser/FunctionParserTest.php

```diff
@@ -201,4 +216,50 @@ public function shouldNotParseFunctionByExplicitValueDefinition(): void
             }
         }
     }
+
+    /**
+     * @test
+     */
+    public function shouldNotParseFunctionDefinedBadly(): void
+    {
+        // ?fullName=Jon,Snow
+        $column = 'fullName';
+        $value = 'Jon,Snow';
+        $this->parser->setQueryParameters([$column => $value]);
+
+        $this->assertTrue($this->parser->supports($column, $value));
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Explicit function definition must have a tuple value.');
+
+        foreach ($this->parser->parse($column, $value) as $filter) {
+            // just iterate through
+            continue;
+        }
+
+        $this->fail('This should not be reached');
```

Comment:
You shouldn't need this line - the testcase will fail anyway if exception is not raised.